### PR TITLE
New Presence, Diziroll ( diziroll.com )

### LINKED ISSUE #3229: diziroll.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months?
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** ZyuzAx#8780

**What's the name of the service?** www.diziroll.com | diziroll

**What should the Presence display?** Dizi/Film isimleri, Dizi/Film resimleri, Dizi/Film süreleri.

**If possible, please provide a logo for the service (512x512 minimum)** 
diziroll

### PR DESCRIPTION
**Pull request by erdem85**
closes #3229, proof that it is working

https://user-images.githubusercontent.com/41632465/110453436-34f92800-80d7-11eb-9a56-1dfc8fcf2798.mp4

=> https://www.diziroll.com/

### CODE REVIEW

File: websites/D/Diziroll/presence.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
+
```
Comment (SlowLife1661) [inline websites/D/Diziroll/presence.ts:26]: Use built-in function, `presence.getTimestamps()`

Comment (SlowLife1661) [reply websites/D/Diziroll/presence.ts:26]: So just remove this and use it on line 66th

File: websites/D/Diziroll/presence.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            presenceData.smallImageText = "Durduruldu";
+          } else {
+            const timestamps = getTimestamps(
+              Math.floor(stream?.currentTime),
+              Math.floor(stream?.duration)
+            );
+      
+            presenceData.smallImageKey = "play";
+            presenceData.smallImageText = "Oynatılıyor";
+            presenceData.startTimestamp = timestamps[0];
```
Comment (SlowLife1661) [inline websites/D/Diziroll/presence.ts:72]: Can use ternary operator

File: websites/D/Diziroll/presence.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
+
+presence.on("UpdateData", async () => {
+    const path: string = document.location.pathname,
+    presenceData: PresenceData = {largeImageKey: "diziroll"};
+    if(path.startsWith("/arsiv")){
```
Comment (SlowLife1661) [inline websites/D/Diziroll/presence.ts:30]: Merge const

File: websites/D/Diziroll/presence.ts (hunk 2 above)
Comment (erdem85) [reply websites/D/Diziroll/presence.ts]: i think it is not neccesary

File: websites/D/Diziroll/presence.ts (hunk 1 above)
Comment (erdem85) [reply websites/D/Diziroll/presence.ts]: thanks

File: websites/D/Diziroll/presence.ts (hunk 2 above)
Comment (SlowLife1661) [reply websites/D/Diziroll/presence.ts]: That's what I meant

Comment (SlowLife1661) [reply websites/D/Diziroll/presence.ts]: @erdem85 ^

**Comment by erdem85** [comment]
finally i fixed codacy's ctrlf errors

File: websites/D/Diziroll/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    "url": ["diziroll.com", "www.diziroll.com"],
+    "description": {
+      "en": "Diziroll is a website that you can watch series.",
+      "tr": "Diziroll bir dizi izleme sitesidir."
+    },
+    "service": "Diziroll",
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/hY9GNiw.png",
+    "thumbnail": "https://i.imgur.com/YsL5RgB.jpg",
```
Comment (Bas950) [inline websites/D/Diziroll/dist/metadata.json:15]: why the white border around the logo and its kinda blurry?

File: websites/D/Diziroll/presence.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  }else if(document.getElementById("archive-page")){
+      presenceData.details = "Bir dizi türünü inceliyor: ";
+      presenceData.state = document.querySelector("div.title").innerText;
+      presenceData.startTimestamp = Date.now();
+
+  }else if(document.getElementById("series-page")){
+      presenceData.details = "Bir diziyi inceliyor:";
+      presenceData.state = document.querySelector("div.top > h1").innerText;
+      presenceData.startTimestamp = Date.now();
```
Comment (Bas950) [inline websites/D/Diziroll/presence.ts:38]: just set startTimestamp once.. and I think you need to MathFloor it

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/D/Diziroll/dist/metadata.json (hunk 4 above)
Comment (erdem85) [reply websites/D/Diziroll/dist/metadata.json]: i think, this is better ( https://github.com/PreMiD/Presences/pull/3220/commits/0999b0db4afc67557d8f3eb525ac0ffd5684fe88 )

Comment (erdem85) [reply websites/D/Diziroll/dist/metadata.json]: i changed the rpc's logo
image

File: websites/D/Diziroll/presence.ts (hunk 5 above)
Comment (erdem85) [reply websites/D/Diziroll/presence.ts]: https://github.com/PreMiD/Presences/pull/3220/commits/cf692a8d65eb2dfe1877c4bcc7db3ebd77e6d42b

**Review by Bas950 — APPROVED** [review]

File: websites/D/Diziroll/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.0",
```
Comment (StrikerFRFX) [inline websites/D/Diziroll/dist/metadata.json:2]: Please use the 1.3 schema

File: websites/D/Diziroll/presence.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      presenceData.state = `${document.querySelector("div.select-season > a").innerText ? document.querySelector("div.select-season > a").innerText : "Bulunamadı"}- ${document.querySelector("div.select-episode > a").innerText ? document.querySelector("div.select-episode > a").innerText : "Bulunamadı"}`;
+      presenceData.buttons = [
+        {label: "İzle", url: document.location.href},
+        {label: "Diziyi Görüntüle", url: document.location.origin + "/" +document.location.pathname.split("/")[1]}
+      ];
+      const timestamps = presence.getTimestamps(stream.currentTime ? Math.floor(stream.currentTime) : null, 
+      stream.duration ? Math.floor(stream.duration) : null);
+      presenceData.smallImageKey = stream.paused ? "pause" : "play";
+      presenceData.smallImageText = stream.paused ? "Durduruldu" : "Oynatılıyor";
+      if (stream.paused == false) {
```
Comment (StrikerFRFX) [inline websites/D/Diziroll/presence.ts:45]:
```suggestion
      if (!stream.paused) {
```

File: websites/D/Diziroll/presence.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+      presenceData.details = "Bir sayfaya bakıyor:";
+      presenceData.state = "Hesabım";
+
+  }else if(path == "/"){
+      presenceData.details = "Bir sayfaya bakıyor:";
+      presenceData.state = "Ana Sayfa";
+
+  }else if(document.getElementById("archive-page")){
+      presenceData.details = "Bir dizi türünü inceliyor: ";
+      presenceData.state = document.querySelector("div.title").innerText;
```
Comment (StrikerFRFX) [inline websites/D/Diziroll/presence.ts:28]:
```suggestion
      presenceData.state = document.querySelector("div.title").textContent;
```

Preferred to use textContent, most presences do this :)

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]

**Review by Bas950 — CHANGES_REQUESTED** [review]
Please upload a few new screens of it workin

**Comment by erdem85** [comment]
> Please upload a few new screens of it workin


https://user-images.githubusercontent.com/41632465/111072185-7e3be400-84ea-11eb-9744-641c7e033dce.mp4

@StrikerFRFX @Bas950

**Review by Bas950 — APPROVED** [review]

**Review by StrikerFRFX — APPROVED** [review]

**Pull request merged**